test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -330,8 +331,10 @@ void stopReadersAndWriters(ReaderGroupManager readerGroupManager, String readerG
         if (!FutureHelpers.await(testState.writersComplete)) {
             log.error("Writers stopped with exceptions");
         }
-        if (!FutureHelpers.await(testState.newWritersComplete)) {
-            log.error("Writers stopped with exceptions");
+        if (testState.autoScaleTestFlag.get()) {
```

Comment:
We have two tests here covering 1. autoScale 2. manualScale. Both invoke the same stopReadersandWriters(). We add additional writers only in autoScale test. So only `readWriteAutoScaleFailoverTest` needs to wait upon the corresponding `newWritersComplete` future and not the `readWriteScaleFailoverTest `. This flag is used to differentiate if it is autoScale or scale test . This is set to true by default. In the setup of `readWriteScaleFailoverTest ` am setting it to `false`. 
To conclude it is used so that the test waits upon `newWritersComplete`  only if it is autoScale and not the manual scale test case.